tests/spec/fs.appendFile.spec.js

```diff
@@ -50,6 +50,38 @@ describe('fs.appendFile', function() {
     });
   });
 
+  it('should append without error when explcitly entering encoding and flag options (default values)' , function(done) {
+    var fs = util.fs();
+    var contents = 'This is a file.';
+    var more = ' Appended.';
+
+    fs.appendFile('/myfile', more , {encoding: 'utf8', flag: 'a'}, function(error) {
```

Comment:
Hi just to nitpick here, did you try using promises instead of done()?
For instance, declare var fsPromises = util.fs().promises;
then return fsPromises.appendFile(......) .catch (error => {......})
also, fsPromises.readFile(......) .catch(error => {expect(error)......}), and you don't need done() at the end.

Reply:
Wouldn't that be testing a different function? I'm not sure I understand why this should be done.

Reply:
We could do the test in both ways, it's true.  Using a `done` callback is fine if you want to use callbacks, and if you're doing Promises, not needed.  In this cas @YuechengWu is using callbacks, so this should be OK>

Reply:
That's an interesting bug you found in class!